File: week6/sentimental/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # getting input
    input = get_string("Text: ")
    # applying function defined below
    grade(input)


def grade(input):
    # initializing letters, words, sentences
    let = 0
    word = 1
    sen = 0
    # looping thru all list
    for i in range(len(input)):
        # to collect data for colemen liau index
        if input[i].isalpha():
            let += 1
        elif input[i].isspace():
            word += 1
        elif input[i] == '.' or input[i] == '?' or input[i] == '!':
            sen += 1
    # Applying Colemen Liau Index
    grade = round((0.0588 * let / word * 100) - (0.296 * sen / word * 100) - 15.8)
    # Printing out the grade
    if grade < 1:
        print("Before Grade 1")
    elif grade > 15:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")
    return


if __name__ == "__main__":
    main()

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int main(void)
{
    //getting input
    string s = get_string("Text: ");
    //ints decleration
    int let = 0, word = 1, sen = 0;
    //loop to check each char of string
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        //to check if they're alphabets
        if (isalpha(s[i]))
        {
            let++;
        }
        //to check if it's a word
        else if (isspace(s[i]))
        {
            word++;
        }
        //to check if it's a sentence
        else if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            sen++;
        }
    }
    //applying colemen liau index
    int grade = round((0.0588 * let / word * 100) - (0.296 * sen / word * 100) - 15.8);
    //checking if it's smaller than 1
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    //checking if it's larger than 16
    else if (grade > 15)
    {
        printf("Grade 16+\n");
    }
    //declearing grade
    else
    {
        printf("Grade %i\n", grade);
    }
    return 0;
}